x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/context/alerts_context.tsx

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+export interface IAlertsContext {
+  alertFlyoutVisible: boolean;
+  setAlertFlyoutVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const AlertsContext = React.createContext({} as IAlertsContext);
```

Comment:
This looks like we're breaking the type.
`React.createContext` takes a type like this: `React.createContext<IAlertsContext>({})` and ofcourse that points out that the empty object isn't correct (its properties are missing), so using `as` is dangerous as it makes the compiler _think_ it's safe, but it isn't, and so all the uses of this context are no longer type safe.

IMHO if we want to support missing fields - we should mark them optional usin `?`, but if we just want to support an empty context, then we should change the type so that it uses the `Option` type (from `fp-ts`), or put the context object under an optional key like this:

```
interface IAlertsVisibility {
  alertFlyoutVisible: boolean;
  setAlertFlyoutVisibility: React.Dispatch<React.SetStateAction<boolean>>;
}
export interface IAlertsContext {
  visibility?: IAlertsVisibility
}
```

Otherwise we're actually making it dangerous because developers *think* the compiler is typechecking, bnut it isn't.

also, nit picky, but we don't seem to use the `I` prefix on interfaces in other parts of Kibana. Do we want to standardise on this? I don't care to much, just thought I'd flag it.

Reply:
refactored